src/www/ui/api/Controllers/CopyrightController.php

```diff
@@ -0,0 +1,104 @@
+ * @file
+ * @brief Controller for copyright queries
+ */
+
+namespace Fossology\UI\Api\Controllers;
+
+use Fossology\Lib\Auth\Auth;
+use Fossology\Lib\Dao\UploadDao;
+use Fossology\Lib\Db\DbManager;
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use Psr\Http\Message\ServerRequestInterface;
+
+
+class CopyrightController extends RestController
+{
+  /**
+   * @var ContainerInterface $container
+   * Slim container
+   */
+  protected $container;
+
+  /**
+   * Get query parameter name for copyright filtering
+   */
+  const COPYRIGHT_PARAM = "status";
+
+  /**
+   * @var copyrightHist $copyrightHist
+   * Copyright Histogram object
+   */
+  private $copyrightHist;
+
+  public function __construct($container)
+  {
+    parent::__construct($container);
+    $this->restHelper = $container->get('helper.restHelper');
+    $this->copyrightHist = $this->restHelper->getPlugin('ajax-copyright-hist');
+  }
+
+  /**
+   * Get all copyrights for a particular upload-tree
+   *
+   * @param  ServerRequestInterface $request
+   * @param  ResponseHelper         $response
+   * @param  array                  $args
+   * @return ResponseHelper
+   */
+  public function getFileCopyrights($request, $response, $args)
+  {
+    $uploadPk = $args["id"];
+    $uploadTreeId = $args["itemId"];
+    $query = $request->getQueryParams();
+    $statusVal = true;
+    $returnVal = null;
+    $finalVal=[];
+    if (!array_key_exists(self::COPYRIGHT_PARAM, $query)) {
+      $returnVal = new Info(400, "Bad Request. 'status' is a required query param", InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+    $status = $query[self::COPYRIGHT_PARAM];
+    if ($status == "active") {
+      $statusVal = true;
+    } else if ($status == "inactive") {
+      $statusVal = false;
+    } else {
+      $returnVal = new Info(400, "Bad Request. Invalid query parameter", InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+    try {
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadPk)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if (!$this->dbHelper->doesIdExist($this->restHelper->getUploadDao()->getuploadTreeTableName($uploadPk), "uploadtree_pk", $uploadTreeId)) {
+        $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+      }
+      if ($returnVal !== null) {
+        return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+      }
+      $agentId = $this->copyrightHist->getAgentId($uploadPk, 'copyright_ars');
+      $uploadTreeTableName = $this->restHelper->getUploadDao()->getuploadTreeTableName($uploadPk);
+      list($rows, $iTotalDisplayRecords, $iTotalRecords)  = $this->copyrightHist->getCopyrights($uploadPk, $uploadTreeId, $uploadTreeTableName, $agentId, 'statement', 'active', $statusVal);
```

Comment:
The limit and offset is not read from the headers like in other endpoints and sent to `getCopyrights()`. Please update.